migration_tools.drush.inc

```diff
@@ -248,6 +263,51 @@ function drush_migration_tools_mt_migrate_file_folders($organization) {
 }
 
 /**
+ * Drush command callback for migration-tools-html-file-list.
+ *
+ */
+function drush_migration_tools_html_file_list($directory) {
+  // Set up options
+  $format = drush_get_option('format');
+  $outfile = drush_get_option('outfile');
+  $mask = '/\.htm/';
+
+  switch ($format) {
+    case "csv":
+      $delimiter = ", ";
+      $newline = "\n";
+      break;
+    default:
+      $delimiter = "\t";
+      $newline = "\n";
+  }
+
+  // Fetch file list
+  $options = array();
+  $files = file_scan_directory($directory, $mask, $options);
+
+  // Generate output
+  $output = "Path" . $delimiter . "File Name" . $delimiter . $newline;
+  foreach ($files as $file) {
+    // We only want the path to the file, exclusive of the file name, so we drop the last part of the uri
+    $fullpath = explode("/", $file->uri, -1);
+    $output .= implode("/", $fullpath) . $delimiter;
+
+    $output .= $file->filename . $delimiter;
+    $output .= $newline;
+  }
+
```

Comment:
The $scanned_directory is so that the the file gets created with the name of the directory we passed in.  That way there is no need for outfile declaration.  There is no accidentally overwritten scans becasue we ran the command without changing the outfile name.  You declare the type of file and thats what you get, all properly named.